tools/SetupFlow/DevHome.SetupFlow/ViewModels/QuickstartPlaygroundViewModel.cs

```diff
@@ -430,73 +446,102 @@ public void UpdateProgress(QuickStartProjectProgress progress)
+                // ensures that the file view is back to a known, clean state.
+                GeneratedFileContent = string.Empty;
 
-            IProgress<QuickStartProjectProgress> progress = new Progress<QuickStartProjectProgress>(UpdateProgress);
+                // Temporarily turn off the provider combobox and ensure user cannot edit the prompt for the moment
+                EnableQuickstartProjectCombobox = false;
+                IsPromptTextBoxReadOnly = true;
+                EnableProjectButtons = false;
+                SetupChat();
 
-            var outputFolder = await GetOutputFolder();
-            _outputFolderForCurrentPrompt = outputFolder.Path;
+                IProgress<QuickStartProjectProgress> progress = new Progress<QuickStartProjectProgress>(UpdateProgress);
 
-            _quickStartProjectGenerationOperation = ActiveQuickstartSelection.CreateProjectGenerationOperation(userPrompt, outputFolder, ActivityId);
-            _quickStartProject = await _quickStartProjectGenerationOperation.GenerateAsync().AsTask(progress);
-            _quickStartProjectGenerationOperation = null!;
-            if (_quickStartProject.Result.Status == ProviderOperationStatus.Success)
-            {
-                SetUpFileView();
-                SetupLaunchButton();
-                EnableProjectButtons = true;
-                TelemetryFactory.Get<ITelemetry>().LogCritical("QuickstartPlaygroundGenerateSuccceded", relatedActivityId: ActivityId);
+                var outputFolder = await GetOutputFolder();
+                _outputFolderForCurrentPrompt = outputFolder.Path;
+
+                _quickStartProjectGenerationOperation = ActiveQuickstartSelection.CreateProjectGenerationOperation(userPrompt, outputFolder, ActivityId);
+                _quickStartProject = await _quickStartProjectGenerationOperation.GenerateAsync().AsTask(progress);
+                _quickStartProjectGenerationOperation = null!;
+                if (_quickStartProject.Result.Status == ProviderOperationStatus.Success)
+                {
+                    SetUpFileView();
+                    SetupLaunchButton();
+                    EnableProjectButtons = true;
+                    TelemetryFactory.Get<ITelemetry>().LogCritical("QuickstartPlaygroundGenerateSuccceded", relatedActivityId: ActivityId);
+                }
+                else
+                {
+                    IsErrorViewVisible = true;
+                    ErrorMessage = StringResource.GetLocalized("QuickstartPlaygroundGenerationFailedDetails", _quickStartProject.Result.DisplayMessage);
+                    TelemetryFactory.Get<ITelemetry>().Log(
+                        "QuickstartPlaygroundGenerateFailed",
+                        LogLevel.Critical,
+                        new ProjectGenerationErrorInfo(_quickStartProject.Result.DisplayMessage, _quickStartProject.Result.ExtendedError, _quickStartProject.Result.DiagnosticText),

```

Comment:
NIT: pass in the exception and have the class break it apart into the data it needs.